cli/src/main/scala/com/codacy/analysis/cli/command/analyse/MetricsToolExecutor.scala

```diff
@@ -0,0 +1,59 @@
+package com.codacy.analysis.cli.command.analyse
+import com.codacy.analysis.cli.command.analyse.AnalyseExecutor.MetricsToolExecutorResult
+import com.codacy.analysis.core.model.FileMetrics
+import org.log4s.{Logger, getLogger}
+
+import scala.util.{Failure, Success}
+
+object MetricsToolExecutor {
+
+  private val logger: Logger = getLogger
+
+  def reduceMetricsToolResultsByFile(metricsResults: Seq[MetricsToolExecutorResult]): Seq[MetricsToolExecutorResult] = {
+    metricsResults
+      .groupBy(_.language)
+      .values
+      .flatMap {
+        _.foldLeft(Option.empty[MetricsToolExecutorResult]) {
+          case (
+              Some(metricsExecutorResAcc @ MetricsToolExecutorResult(_, _, Success(fileMetricsAcc))),
+              metricsExecutorRes @ MetricsToolExecutorResult(_, _, Success(fileMetrics))) =>
+            Some(
+              metricsExecutorResAcc.copy(
+                files = metricsExecutorResAcc.files ++ metricsExecutorRes.files,
```

Comment:
Please create variables for these two lines. Having operations mixed inside invocations is always a no go for me because is very sneaky.